src/Bicep.Core/LanguageConstants.cs

```diff
@@ -48,37 +48,32 @@ public static class LanguageConstants
 
         public static readonly string DeclarationTypesString = LanguageConstants.DeclarationTypes.Keys.ConcatString(ListSeparator);
 
-        public static TypeSymbol CreateParameterModifierType(TypeSymbol parameterType)
+        public static TypeSymbol CreateParameterModifierType(TypeSymbol primitiveType, TypeSymbol allowedValuesType)
         {
-            if (parameterType.TypeKind != TypeKind.Primitive)
-            {
-                throw new ArgumentException($"Modifiers are not supported for type '{parameterType.Name}'.");
-            }
-
-            return new NamedObjectType($"ParameterModifier_{parameterType.Name}", CreateParameterModifierProperties(parameterType), additionalPropertiesType: null);
+            return new NamedObjectType($"ParameterModifier<{allowedValuesType.Name}>", CreateParameterModifierProperties(primitiveType, allowedValuesType), additionalPropertiesType: null);
```

Comment:
This change is because the underscore is going to look odd when displaying union types - e.g.:
```
ParameterModifier<a | b | c>
```
vs
```
ParameterModifier_a | b | c
```

Reply:
Makes sense.